src/Vector3.h

```diff
@@ -235,6 +235,12 @@ class Vector3
 		return *this;
 	}
 
+	/** Provides a hash of a vector's contents */
+	size_t operator()(const Vector3<T> & a_Vector) const
```

Comment:
Don't make this part of vector3. Either refactor VectorHasher out of aPath as aseperate object, or create one yourself at global scope.